sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseConfiguration.java

```diff
@@ -0,0 +1,129 @@
+package com.azure.monitor.opentelemetry.exporter.implementation.quickpulse;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.logging.ClientLogger;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+
+public class QuickPulseConfiguration {
+    private static final ClientLogger logger = new ClientLogger(QuickPulseDataFetcher.class);
+    private  AtomicReference<String> etag = new AtomicReference<>();
+    private ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> metrics = new ConcurrentHashMap<>();
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    public synchronized String getEtag() {
+        return this.etag.get();
+    }
+
+    public synchronized void setEtag(String etag) {
+        this.etag.set(etag);
+    }
+
+    public synchronized ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> getMetrics() {
+        return this.metrics;
+    }
+
+    public synchronized void setMetrics(ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> metrics) {
+        this.metrics = metrics;
+    }
+
+    public synchronized void updateConfig(String etagValue, ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> otelMetrics) {
```

Comment:
I have left some other comments that note that the response can contain derived metrics that are of different telemetry types. With that in mind, leaving a suggestion on how to alter this method:

- don't take the concurrent hash map as a parameter
- instead of calling this.setMetrics, I would call this.parseMetrics here (though I would name this parseDerivedMetricInfos) to update the map.

Reply:
Since multiple threads are accessing and modifying the etag and derived metrics in QuickPulseConfiguration, both objects need a lock when either one gets modified. Updating metrics separately in parseDerivedMetrics() would not work.